tunix/generate/utils.py

```diff
@@ -410,141 +415,122 @@ def transfer_state_with_mappings(
   Returns:
     The target state with the transferred values.
   """
+  tgt_flat_list = dst_state.flat_state()
+  sharding_dict = None
+  if reshard_fn:
```

Comment:
you have if reshard_fn logic in L522, maybe move L421 in the if check of L522 as well to move relevant logic close